packages/dev/core/src/Materials/Background/backgroundMaterial.ts

```diff
@@ -970,6 +970,9 @@ export class BackgroundMaterial extends PushMaterial {
 
         if (scene.performancePriority !== ScenePerformancePriority.BackwardCompatible) {
             this.checkReadyOnlyOnce = true;
+            scene.onScenePerformancePriorityChangedObservable.addOnce(() => {

```

Comment:
Shouldn't `onScenePerformancePriorityChangedObservable` be cleaned up from the observer we added when the the material is disposed? Otherwise, we will have memory leak.

Same comments for NodeMaterial / PBRBaseMaterial / ...

Reply:
of course, it should. thanks :-)